src/js/relayclient/ServerHelper.js

```diff
@@ -186,51 +201,92 @@ class ServerHelper {
       this.fromBlock = fromBlock
       await this.fetchRelaysAdded()
     }
-    return this.createActiveRelayPinger(this.filteredRelays, this.httpSend, gasPrice, this.verbose)
+    return this.createActiveRelayPinger(this.filteredRelays, this.httpSend, gasPrice, this.verbose, this.preferredRelays)
   }
 
-  createActiveRelayPinger (filteredRelays, httpSend, gasPrice, verbose) {
-    return new ActiveRelayPinger(filteredRelays, httpSend, gasPrice, verbose)
+  createActiveRelayPinger (filteredRelays, httpSend, gasPrice, verbose, preferredRelays) {
+    let pinger = new ActiveRelayPinger(filteredRelays, httpSend, gasPrice, verbose)
+    if (preferredRelays) {
+      const prefs = this.preferredRelays.map(relayUrl => ({
+        relayUrl,
+        transactionFee: 0,
+        stake: PREFERRED_STAKE,
+        unstakeDelay: PREFERRED_DELAY
+      })
+      )
+      // if there are preferred relays, create a pinger that uses them FIRST, and only
+      // falls-back to the "normal" pinger, based on the filtered list.
+      pinger = new ActiveRelayPinger(prefs, httpSend, gasPrice, verbose, pinger)
```

Comment:
Create a new object. But wait... We didn't check for preferredRelays. Let's create a different one...

Reply:
please explain your pain-point, and suggest a change.

Reply:
``` if (preferredRelays) {
    return new ActiveRelayPinger(bla)
} else {
    return new ActiveRelayPinger(blabla)
}```
makes much more sense. I don't understand the need to wrap one ActiveRelayPinger inside another ActiveRelayPinger,

Reply:
but that's not the case. its not either X or Y:
 its " ActiveRelayPinger(X), and if they all fail, ActiveRelayPinger(Y)."

I can achieve this either by a "wrapper" ActiveRelayPinger that do it, or that every pinger can have a "next" pinger to fallback to it, once all the relays it manages by itself fail.